app/components/CollectionPreview.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+import * as React from "react";
+import { Link } from "react-router-dom";
+import styled, { withTheme } from "styled-components";
+import Collection from "models/Collection";
+import CollectionIcon from "components/CollectionIcon";
+import Flex from "components/Flex";
+import Highlight from "components/Highlight";
+import CollectionMenu from "menus/CollectionMenu";
+
+type Props = {
+  collection: Collection,
+  highlight?: ?string,
+};
+
+class CollectionPreview extends React.Component<Props> {
```

Comment:
Lets make this a functional component, trying to move towards them for new stuff.